bugsnag-plugin-android-anr/src/main/java/com/bugsnag/android/AnrPlugin.kt

```diff
@@ -46,27 +67,26 @@ internal class AnrPlugin : Plugin {
      * Notifies bugsnag that an ANR has occurred, by generating an Error report and populating it
      * with details of the ANR. Intended for internal use only.
      */
-    @Suppress("UNUSED_PARAMETER")
     private fun notifyAnrDetected(info: Long, userContext: Long) {
-        val thread = Looper.getMainLooper().thread
-
         // generate a full report as soon as possible, then wait for extra process error info
-        val exc = RuntimeException()
-        exc.stackTrace = thread.stackTrace
+        var stackTrace = Looper.getMainLooper().thread.stackTrace
+        var nativeTrace: List<NativeStackframe>? = null
 
         @Suppress("UNCHECKED_CAST")
         val clz = Class.forName("com.bugsnag.android.NdkPlugin") as Class<Plugin>
```

Comment:
I missed this on the last review, but it'd probably be worth wrapping the whole native stacktrace generation with a `try-catch`, as JVM reflection is brittle and prone to throwing exceptions. If an exception does occur at least we would still report something rather than terminating the process.

Reply:
OK, I've wrapped the whole thing in a try-catch